### PR TITLE
Initial support for OTP 26

### PR DESCRIPTION
**Pull request by martinsumner**
Add support for OTP 26.

Fixes some type issues newly discovered by improved dialyzer, and also keeps dialyzer happy by moving functions and includes only used in test, into the ifdef(TEST).

### CODE REVIEW

File: src/leveled_cdb.erl [hunk 1]
```diff
@@ -1883,10 +1895,12 @@ create(FileName,KeyValueList) ->
 
 %% Should not be used for non-test PUTs by the inker - as the Max File Size
 %% should be taken from the startup options not the default
-put(FileName, Key, Value, {LastPosition, HashTree}) ->
-    put(FileName, Key, Value, {LastPosition, HashTree},
-            ?BINARY_MODE, ?MAX_FILE_SIZE, false).
-
+put(FileName, Key, Value, {LastPosition, HashTree}) when is_list(FileName) ->
+    file_put(FileName, Key, Value, {LastPosition, HashTree},
```
Comment (ThomasArts) [inline src/leveled_cdb.erl:1899]: using `file_put`here and therefore correct

File: src/leveled_iclerk.erl [hunk 2]
```diff
@@ -863,7 +861,7 @@ filter_output_fun(FilterFun, FilterServer, MaxSQN, Strategy) ->
 
 -spec to_retain(leveled_codec:journal_key(),
                 leveled_inker:filterfun(),
-                leveled_inker:fillter_server(),
+                leveled_inker:filterserver(),
```
Comment (ThomasArts) [inline src/leveled_iclerk.erl:864]: nice catch... is that improved dialyzer?

**Review by ThomasArts — APPROVED** [review]
Looks good to me

File: src/leveled_iclerk.erl (hunk 2 above)
Comment (martinsumner) [reply src/leveled_iclerk.erl:864]: Yes. OTP26 now seems to detect lots of "obvious" problems that weren't revealed before - in particular mis-typed type references, and references to unexported types.

**Pull request merged**